homeassistant/components/tplink/switch.py

```diff
@@ -34,8 +34,8 @@ async def async_setup_entry(
     if device.is_strip:
         # Historically we only add the children if the device is a strip
         _LOGGER.debug("Initializing strip with %s sockets", len(device.children))
-        for child in device.children:
-            entities.append(SmartPlugSwitch(child, coordinator))
+        for i, child in enumerate(device.children):
+            entities.append(SmartPlugSwitch(child, coordinator, i))
```

Comment:
I'm not sure if it's a good idea to hard code the children names instead of using the device provided ones, any specific reason for doing that?